src/config.c

```diff
@@ -1835,6 +1867,38 @@ static void stringConfigRewrite(typeData data, const char *name, struct rewriteC
     rewriteConfigStringOption(state, name,*(data.string.config), data.string.default_value);
 }
 
+/* SDS Configs */
+static void sdsConfigInit(typeData data) {
+    *data.sdsd.config = (data.sdsd.default_value || !data.sdsd.convert_empty_to_null) ? sdsnew(data.sdsd.default_value) : NULL;
+}
+
+static int sdsConfigSet(typeData data, sds value, int update, char **err) {
+    if (data.sdsd.is_valid_fn && !data.sdsd.is_valid_fn(value, err))
+        return 0;
+    sds prev = *data.sdsd.config;
+    *data.sdsd.config = (value || !data.sdsd.convert_empty_to_null) ? sdsdup(value) : NULL;
```

Comment:
I would flip this logic back so that it's easier to read, as per my other comment.
```
data.sdsd.convert_empty_to_null && !value ? NULL : sdsdup(value); 
```

Reply:
Note the value[0] in either case. We're checking to see if it's an empty string, not a NULL value.

Reply:
I can certainly check for `value[0]`. However I would still do a check for `value` so that we won't get a segfault if the input argument `value` is NULL.

Reply:
And for sds input, I think I will do `sdslen(value) == 0` instead

Reply:
Ok, sure.